alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Ledger.hs

```diff
@@ -0,0 +1,195 @@
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PatternSynonyms #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE StandaloneDeriving #-}
+{-# LANGUAGE TypeApplications #-}
+{-# LANGUAGE TypeFamilies #-}
+{-# LANGUAGE UndecidableInstances #-}
+
+module Cardano.Ledger.Alonzo.Rules.Ledger
+
+{-
+  ( LEDGER,
+    LedgerEnv (..),
+    LedgerPredicateFailure (..),
+    PredicateFailure,
+  )
+
+-}
+
+where
+
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Era (Crypto, Era)
+import Control.State.Transition
+  ( Assertion (..),
+    AssertionViolation (..),
+    Embed (..),
+    STS (..),
+    TRC (..),
+    TransitionRule,
+    judgmentContext,
+    trans,
+  )
+import Data.Kind(Type)
+import Data.Sequence (Seq)
+import Data.Sequence.Strict (StrictSeq)
+import qualified Data.Sequence.Strict as StrictSeq
+import GHC.Records (HasField, getField)
+import Shelley.Spec.Ledger.BaseTypes (ShelleyBase)
+import Shelley.Spec.Ledger.Coin (Coin)
+import Shelley.Spec.Ledger.EpochBoundary (obligation)
+import Shelley.Spec.Ledger.Keys (DSignable, Hash)
+import Shelley.Spec.Ledger.LedgerState
+  ( DPState (..),
+    DState (..),
+    PState (..),
+    UTxOState (..),
+  )
+import Shelley.Spec.Ledger.STS.Delegs (DELEGS, DelegsEnv (..), DelegsPredicateFailure)
+import Shelley.Spec.Ledger.STS.Utxo
+  ( UtxoEnv (..),
+  )
+import Shelley.Spec.Ledger.STS.Ledger(LedgerEnv(..), LedgerPredicateFailure(..))
+import Shelley.Spec.Ledger.TxBody (DCert, EraIndependentTxBody)
+import Cardano.Ledger.Shelley.Constraints(PParamsDelta)
+
+import Cardano.Ledger.Alonzo.Tx(Tx(..),IsValidating(..))
+import Cardano.Ledger.Alonzo.Rules.Utxow(AlonzoUTXOW,AlonzoPredFail)
+
+-- =======================================
+
+-- | The uninhabited type that marks the (STS Ledger) instance in the Alonzo Era.
+data AlonzoLEDGER era
+
+
+-- | An abstract Alonzo Era, Ledger transition. Fix 'someLedger' at a concrete type to
+--   make it concrete. Depends only on the "certs" and "isValidating" HasField instances.
+ledgerTransition ::
+  forall (someLEDGER:: Type -> Type ) era.
+  ( Signal (someLEDGER era) ~ Core.Tx era,
+    State (someLEDGER era) ~ (UTxOState era, DPState (Crypto era)),
+    Environment (someLEDGER era) ~ LedgerEnv era,
+    Embed (Core.EraRule "UTXOW" era) (someLEDGER era),
+    Embed (Core.EraRule "DELEGS" era) (someLEDGER era),
+
+    Environment (Core.EraRule "DELEGS" era) ~ DelegsEnv era,
+    State (Core.EraRule "DELEGS" era) ~ DPState (Crypto era),
+    Signal (Core.EraRule "DELEGS" era) ~ Seq (DCert (Crypto era)),
+
+    Environment (Core.EraRule "UTXOW" era) ~ UtxoEnv era,
+    State (Core.EraRule "UTXOW" era) ~ UTxOState era,
+    Signal (Core.EraRule "UTXOW" era) ~ Core.Tx era,
+
+    HasField "certs" (Core.TxBody era) (StrictSeq (DCert (Crypto era))),
+    HasField "isValidating" (Core.Tx era) IsValidating,
+    Era era
+  ) => TransitionRule (someLEDGER era)
+ledgerTransition = do
```

Comment:
this is only covering the `txvaltag tx = True` case of the rule (ie equation 5 of figure 15 of the spec). we still need the False case.